articles/quickstart/native/ionic3/01-login.md

```diff
@@ -0,0 +1,262 @@
+---
+title: Login
+default: true
+description: This tutorial demonstrates how to add authentication and authorization to an Ionic 3 app
+budicon: 448
+---
+
+<%= include('../../../_includes/_package', {
+  org: 'auth0-samples',
+  repo: 'auth0-ionic3-samples',
+  path: '01-Login',
+  requirements: [
+    'Ionic 3.x',
+    'Angular 5+'
+  ]
+}) %>
+
+<%= include('../_includes/_ionic_setup') %>
+
+## Set Up URL Redirects
+
+Use the `onRedirectUri` method from **auth0-cordova** when your app loads to properly handle redirects after authentication.
+
+```js
+// src/app/app.component.ts
+
+import { Component } from '@angular/core';
+import { Platform } from 'ionic-angular';
+import { StatusBar } from '@ionic-native/status-bar';
+import { SplashScreen } from '@ionic-native/splash-screen';
+
+import { TabsPage } from '../pages/tabs/tabs';
+
+// Import Auth0Cordova
+import Auth0Cordova from '@auth0/cordova';
+
+@Component({
+  templateUrl: 'app.html'
+})
+export class MyApp {
+  rootPage:any = TabsPage;
+
+  constructor(
+    platform: Platform,
+    statusBar: StatusBar,
+    splashScreen: SplashScreen
+  ) {
+    platform.ready().then(() => {
+      // Okay, so the platform is ready and our plugins are available.
+      // Here you can do any higher level native things you might need.
+      statusBar.styleDefault();
+      splashScreen.hide();
+
+      // Redirect back to app after authenticating
+      (window as any).handleOpenURL = (url: string) => {
+        Auth0Cordova.onRedirectUri(url);
+      }
+    });
+  }
+}
+```
+
+## Configure Auth0
+
+Create a new file called `auth.config.ts` in your `src/services` folder to provide the necessary Auth0 configuration for your Ionic app:
+
+```js
+// src/services/auth.config.ts
+
+export const AUTH_CONFIG = {
+  // Needed for Auth0 (capitalization: ID):
+  clientID: '${account.clientId}',
+  // Needed for Auth0Cordova (capitalization: Id):
+  clientId: '${account.clientId}',
+  domain: '${account.namespace}',
+  callbackURL: location.href,
+  packageIdentifier: 'YOUR_PACKAGE_ID' // config.xml widget ID, e.g., com.auth0.ionic
```

Comment:
Can we add, this value should be the same as the uri you specified for callback url?

Reply:
Which callback URL are you referring to? Since there are actually a couple... (e.g., the `location.href` above, the Client settings URL which is `[PACKAGE_ID]://[DOMAIN]/cordova/[PACKAGE_ID]/callback`

I think just saying this introduces more confusion on what we're referring to. Can you clarify?

Reply:
I meant it for line 77, that value should always be the one you used when installing custom-uri-scheme, you shouldn't need to specify line 76.

Reply:
👍